### ISSUE TITLE
warning CA1068: Method should take CancellationToken as the last parameter

### ISSUE THREAD

**Issue by zotrb** [dotnet/roslyn-analyzers#7334]
### Analyzer

**Diagnostic ID**: [CA1068](https://learn.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/quality-rules/ca1068)

### Describe the improvement

In methods that take `params` parameters, the `params` parameter must be last, so it is not possible to comply with this rule.

### Describe suggestions on how to achieve the rule

This warning should not be raised if a method uses a `params` parameter.

**Comment by buyaa-n** [comment]
I do not see [CA1068](https://learn.microsoft.com/en-gb/dotnet/fundamentals/code-analysis/quality-rules/ca1068) when `params` 
used for the last parameter, CA1068 only raises when the `params` removed from last parameter which is expected. Not sure if it is reproducible in older versions @zotrb which analyzer version are you using? Closing for now with can't repro

```cs
internal static void Example(string? value, CancellationToken ct, params int[] parameters)
{
}
```

**Comment by zotrb** [comment]
Hi,
Thanks for investigating.

I reported this a few versions back.
I’ve just tried this on VS 17.11.2 in a dotnet 8 C# project and I can’t reproduce the problem, so it looks as though it’s already been fixed.

Best regards,
Jason



Jason Williams
Application Developer Team Leader

T: +44 (0) 1908 650 098
E: ***@***.******@***.***>

Red Bull Technology
Building 1, Bradbourne Drive
Tilbrook, Milton Keynes, UK
MK7 8BJ

[cid:orbr_email_signature_817x160rb20launch_1ed97cb9-529e-44de-a8f1-85f93a25c249.jpg]<http://www.redbullracing.com/>

Red Bull Technology Limited is a company registered in England and Wales under number 05202976 and whose registered office is Building 1, Bradbourne Drive, Tilbrook, Milton Keynes, MK7 8BJ
Please note, Red Bull Technology Limited only enter into contracts with suppliers based on Red Bull Technology Terms and Conditions of Purchase, copies of which are available on request. This e-mail is confidential and may contain valuable information. If you are not the intended recipient, you should not copy it, re-transmit it, use it or disclose its contents, but should return it to the sender immediately and delete your copy from your system.
